pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -138,6 +138,12 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logge
 			return fmt.Errorf("failed to wait for created ProwJob to appear in cache: %w", err)
 		}
 
+		// Before updating, re-get the object in case it was modified
+		prpqr := &v1.PullRequestPayloadQualificationRun{}
+		if err := r.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: req.Namespace, Name: req.Name}, prpqr); err != nil {
+			return fmt.Errorf("failed to get the PullRequestPayloadQualificationRun: %s in namespace %s: %w", req.Name, req.Namespace, err)
+		}
+
```

Comment:
This will not fix the race,there is still a window between the get and the update. In this case, I think we need to act on the conflict error after update, perhaps in a backoff. 
- Try to update
- Check for errors, if conflict, get and retry
- If errr but not a conflict, a real error
- If too many retries, a real error